merge AccountScriptingInterface into GlobalServicesScriptingInterface

dback2 opened the pull request:
- move isLoggedIn, logOut, loggedIn (including associated member flag variable), and checkAndSignalForAccessToken functions from Account to Global Services API and wrap existing calls from Account API to their associated Global Services API functions
- add loggedInChanged to Global Services API as well as a wrapper function onUsernameChanged to mimic the loggedInChanged behavior from Account API as well as to call myUsernameChanged signal as it was directly doing in Global Services API via AccountManager::usernameChanged listener before
- remove checkAndSignalForAccessToken, usernameChanged, loggedInChanged, and onUsernameChanged from Account API as nothing was using them there
- point Account context to GlobalServicesScriptingInterface instead of AccountScriptingInterface to effectively sever all direct uses of Account API

In later PRs:
-Delete AccountScriptingInterface class completely.
-Rename GlobalServicesScriptingInterface to AccountScriptingInterface (we have decided since Account is more used than GlobalServices that Account naming will be used instead for the long term).
-Remove the GlobalServices contexts (at a set date after advanced notice of removal).

Test Plan:

https://highfidelity.testrail.net/index.php?/cases/view/7536 (Desktop)
https://highfidelity.testrail.net/index.php?/cases/view/7535 (HMD)

These changes are only a behind the scenes refactor and not a functionality change. This should cause no visible differences in functionality for anything account related such as logging in/out, checking login status, accessing username, using Marketplace, etc.

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/4400/HighFidelity-Beta-PR11669-cffcf2d8df42c9c7e22c5fa75c08f27d74cd9e86.dmg

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/4400/HighFidelity-Beta-PR11669-cffcf2d8df42c9c7e22c5fa75c08f27d74cd9e86.exe

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/4403/HighFidelity-Beta-PR11669-c8a0f9761a5670c370b1f6bb331f5718b9b64637.exe

ZappoMan commented:
One side effect of this change is that all the `Q_INVOKABLE` and `Q_PROPERTY` members of `GlobalServicesScriptingInterface ` will not be exposed to JS as methods of `Account` is that a good thing?

dback2 commented:
Why would they not exposed if Account context points to GlobalServicesScriptingInterface?

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/highfidelity/hifi?pullRequest=11669) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/highfidelity/hifi?pullRequest=11669) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: dback2<br/>:x: David Back<br/><hr/>**David Back** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/highfidelity/hifi?pullRequest=11669) it.</sub>

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5374/HighFidelity-Beta-PR11669-3873a8f91ef4be8fbee47111e5e9a4dbef66e46b.dmg

marko8904 commented:
build this please

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5383/HighFidelity-Beta-PR11669-3873a8f91ef4be8fbee47111e5e9a4dbef66e46b.exe

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5383/HighFidelity-Beta-PR11669-3873a8f91ef4be8fbee47111e5e9a4dbef66e46b.dmg

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5426/HighFidelity-Beta-PR11669-5e1064800a1e5eec5dff0bb84eec8a868d53ded5.exe

dback2 commented:
build this please

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5434/HighFidelity-Beta-PR11669-5e1064800a1e5eec5dff0bb84eec8a868d53ded5.exe

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5434/HighFidelity-Beta-PR11669-5e1064800a1e5eec5dff0bb84eec8a868d53ded5.dmg

Nex-Pro commented:
Testing Win 10

Nex-Pro commented:
Stopped part way through test rail. Unable to connect to dev-content.

Also IMO test rail needs breaking into HMD and desktop.  It also has two links to other suites of tests embedded in it. @dback2  Was fine appart from some minor inaccuracies in the expected results up to that point.

https://highfidelity.testrail.net/index.php?/tests/view/24288&group_by=cases:section_id&group_order=asc&group_id=1252 it is marked as pass but is blocked.

dback2 commented:
Split test plans into HMD and Desktop.
Fixed links to the 2 external test plans to regress.
Adjusted test plan to use hifiqa-rc instead of Avatar Island.
Should be ready to test again.

thoys commented on interface/src/scripting/GlobalServicesScriptingInterface.cpp:
```diff
@@ -45,8 +48,29 @@ GlobalServicesScriptingInterface* GlobalServicesScriptingInterface::getInstance(
+    if (accountManager->isLoggedIn()) {
+        return accountManager->getAccountInfo().getUsername();
+    }
+    else {
```
small style issue: `} else {`
https://docs.highfidelity.com/build-guide/coding-standards#4-1-5-the-if-else-class-of-statements-should-have-the-following

cain-kilgore commented:
Testing.

cain-kilgore commented:
# Test Failed
### [Test Run](https://highfidelity.testrail.net/index.php?/runs/view/1746&group_by=cases:section_id&group_order=asc)

hifi-gustavo commented:
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/5769/app_11669_PR.apk

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5769/HighFidelity-Beta-PR11669-be736373f29b77e24429eed651f2bd2b75f82730.exe

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5769/HighFidelity-Beta-PR11669-be736373f29b77e24429eed651f2bd2b75f82730.dmg

dback2 commented:
I think we probably need to hold for hifiqa-rc update as it's the only dev domain that I know of where avatar wearable checkout can currently be tested.

dback2 commented:
Checkout zone added to dev-content at around 80,-10,10 and test plan adjusted.

hifi-gustavo commented:
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/5845/app_11669_PR.apk

dback2 commented:
Not sure what happened to the original test plans but instead of going to hifiqa-rc you can now go to dev-content/80,-10,10 and use the checkout zone there to test checkout of wearables instead.

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5845/HighFidelity-Beta-PR11669-ab5b238aea1be0813a25e095359a845527a85044.exe

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5845/HighFidelity-Beta-PR11669-ab5b238aea1be0813a25e095359a845527a85044.dmg

sbennett77 commented:
Currently testing

sbennett77 commented:
Testing Passed.

dback2 commented:
Test plan links fixed

thoys commented:
This will need a re-test, correctly updated test plans are PR description.

QAadahlin commented:
Currently testing

QAadahlin commented:
@conklin94122 fails

QAadahlin commented:
@conklin94122 fails

When you attempt to change the security picture, it hangs in "submitting" limbo and never finishes loading.

QAadahlin commented:
@conklin94122 - fails
When you attempt to change the security picture, it hangs in "submitting" limbo and never finishes loading.
![securitypic](https://user-images.githubusercontent.com/35040758/34540028-3aa92712-f090-11e7-9648-8a0d5e9fdf3d.JPG)
If you press the cancel button in the "change security picture" , then go to "change passphrase", the following image is what you see. 
![wallet](https://user-images.githubusercontent.com/35040758/34540085-6ec8699a-f090-11e7-965a-9f898628a47d.JPG)
When you go to change the security passphrase first, it takes. You can then go to attempt to change your security picture (which again stays in "submitting" limbo) without seeing the latter defect.

dback2 commented:
@QAadahlin @conklin94122 please verify this failure does not exist on master build

conklin94122 commented:
It didn't fail in a different PR build a few minutes ago but I'll check master now to be safe.

QAadahlin commented:
Testing on latest build

QAadahlin commented:
@conklin94122 #11669 PASSED on the latest build (build #7616).

dback2 commented:
PR notes for 7616 specifically call out the above issue

conklin94122 commented:
@dback2 Can you rebase this PR against current master and we can re-retest?

dback2 commented:
@conklin94122 done build incoming :)

hifi-gustavo commented:
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/5927/app_11669_PR.apk

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5927/HighFidelity-Beta-PR11669-6c8987d75fc50dfd52b90b6a0b8053c0a942c2fc.dmg

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5927/HighFidelity-Beta-PR11669-6c8987d75fc50dfd52b90b6a0b8053c0a942c2fc.exe

hifi-gustavo commented:
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/5928/app_11669_PR.apk

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5928/HighFidelity-Beta-PR11669-6c8987d75fc50dfd52b90b6a0b8053c0a942c2fc.exe

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5928/HighFidelity-Beta-PR11669-6c8987d75fc50dfd52b90b6a0b8053c0a942c2fc.dmg

QAadahlin commented:
@conklin94122 Re-testing build

QAadahlin commented:
@conklin94122 BLOCKED due to inability to access dev-content/80,-10,10, hifi://empty, or hifi://hifiqa-rc. Previous bug has been fixed and passed.

dback2 commented:
@conklin94122 @QAadahlin Try dev-content/80,-10,10 again you should now have access

conklin94122 commented:
Thanks much, @dback2

QAadahlin commented:
@conklin94122 Currently retesting

QAadahlin commented:
@conklin94122 Fails. You cannot edit, remove, or purchase the wearable object. When you select the object. The only way to remove the item was to leave the domain. 
![screen shot 2018-01-04 at 1 04 48 pm copy](https://user-images.githubusercontent.com/35040758/34582557-7b5e1f4e-f151-11e7-9c5b-a0952a318584.jpg)

dback2 commented:
Rebasing for protocol change - ready for retest on new build

hifi-gustavo commented:
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/5949/app_11669_PR.apk

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5949/HighFidelity-Beta-PR11669-81f3590954d26350da0208c15e49d0fce1c2c75d.exe

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5949/HighFidelity-Beta-PR11669-81f3590954d26350da0208c15e49d0fce1c2c75d.dmg

QAadahlin commented:
@conklin94122 Re-testing with new build

QAadahlin commented:
@conklin94122 No longer testing #11669

hifi-gustavo commented:
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/5968/app_11669_PR.apk

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5968/HighFidelity-Beta-PR11669-ac8b8da7b1ab0c0dc3926b696f863f25e7a4c76e.exe

conklin94122 commented:
build this please

hifi-gustavo commented:
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/5986/app_11669_PR.apk

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5986/HighFidelity-Beta-PR11669-ac8b8da7b1ab0c0dc3926b696f863f25e7a4c76e.exe

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5986/HighFidelity-Beta-PR11669-ac8b8da7b1ab0c0dc3926b696f863f25e7a4c76e.dmg

QAadahlin commented:
@conklin94122 Currently re-testing new build

sbennett77 commented:
Currently Testing.

QAadahlin commented:
@conklin94122 Testing complete. PASSED.

conklin94122 commented:
QA approved but it has conflicts. Please let me know if the changes invalidate testing and we need to retest, @dback2

hifi-gustavo commented:
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/6013/app_11669_PR.apk

hifi-gustavo commented:
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/6014/app_11669_PR.apk

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/6014/HighFidelity-Beta-PR11669-76f278f8998bdb70f1916047ae157fb6205595c4.dmg

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/6014/HighFidelity-Beta-PR11669-76f278f8998bdb70f1916047ae157fb6205595c4.exe
